### PR TITLE
fix(tracing): Use deprecated tracing origins option if set

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix

## :scroll: Description

Until the next major version, where we can remove the deprecated options, there are 3 ways to set trace propagation targets. The oldest `tracingOrigins` option was not working as the default value of the newer option always overwrote it.

## :green_heart: How did you test it?
sample app, unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: test/tracing/reactnativetracing.test.ts [hunk 1]
```diff
@@ -117,6 +119,98 @@ describe('ReactNativeTracing', () => {
     jest.clearAllMocks();
   });
 
+  describe('trace propagation targets', () => {
```
Comment (Lms24) [inline test/tracing/reactnativetracing.test.ts:122]: l/recommendation: Afaik, in the JS SDK we have tests that show which option is taken if multiple of these properties are specified. Admittedly it's an edge case, so feel free to disregard (not a blocker).

**Review by Lms24 — APPROVED** [review]
LGTM, one optional suggestion.
